pytim/gitim.py

```diff
@@ -199,24 +202,19 @@ def alpha_shape(self, alpha, group, layer):
             extraids = np.append(extraids, [-1] * n_cube)
         else:
             n_cube = 0
-            extrapoints = np.copy(points)
+            extrapoints = points
             extraids = np.arange(len(points), dtype=np.int)
 
-        self.triangulation.append(Delaunay(extrapoints))
-        try:
-            triangulation = self.triangulation[layer]
-        except IndexError:
-            raise IndexError("alpha_shape called using a wrong layer")
-        triangulation.radii = np.append(group.radii[extraids[extraids >= 0]],
-                                        np.zeros(8))
+        triangulation = Delaunay(extrapoints)
+        self.triangulation.append(triangulation)
```

Comment:
This will fail if alpha_shape is called with a specific layer: it assumes somehow that alpha_shape() is always called within the same loop over all layers.

Reply:
Is this only when a user calls a specific layer?

Reply:
as far as I can see, yes